File: 250224/재귀함수의 꽃/the-flower-of-the-recursive-function.java
import java.util.*;
import java.io.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringBuilder sb = new StringBuilder();
    static int n;

    static void printCnt(int n) {
        if (n == 0)
            return;
        sb.append(n).append(" ");
        printCnt(n - 1);
        sb.append(n).append(" ");
    }

    public static void main(String[] args) throws IOException {
        n = Integer.parseInt(br.readLine());
        printCnt(n);
        System.out.println(sb);

    }
}

File: 250224/재귀함수의 꽃/the-flower-of-the-recursive-function.cpp
#include <iostream>

using namespace std;

void printCnt(int n) {
    if (n == 0)
        return;
    cout << n << " ";
    printCnt(n - 1);
    cout << n << " ";
}

int main() {
    int n;
    cin >> n;

    printCnt(n);
    return 0;
}